File: src/main/java/cc/hyperium/mixins/renderer/MixinRenderManager.java
package cc.hyperium.mixins.renderer;

import net.minecraft.client.renderer.entity.RenderManager;
import net.minecraft.client.renderer.entity.RenderPlayer;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

import java.util.Map;

@Mixin(RenderManager.class)
public interface MixinRenderManager {
    @Accessor
    double getRenderPosX();

    @Accessor
    double getRenderPosY();

    @Accessor
    double getRenderPosZ();

    @Accessor
    Map<String, RenderPlayer> getSkinMap();
}

File: src/main/kotlin/cc/hyperium/mixins/renderer/MixinRenderManager.kt
package cc.hyperium.mixins.renderer

import net.minecraft.client.renderer.entity.RenderManager
import net.minecraft.client.renderer.entity.RenderPlayer
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Accessor

@Mixin(RenderManager::class)
interface MixinRenderManager {

    @Accessor
    fun getRenderPosX(): Double

    @Accessor
    fun getRenderPosY(): Double

    @Accessor
    fun getRenderPosZ(): Double

    @Accessor
    fun getSkinMap(): Map<String, RenderPlayer>

}

val RenderManager.mixinType: MixinRenderManager
    get() = this as MixinRenderManager